### ISSUE TITLE
Improved documentation

### ISSUE THREAD

**Issue by tkashkin** [tkashkin/GameHub#270]
GameHub currently doesn't have documentation and some features may be confusing.

**Comment by hlechner** [comment]
I'm looking to document the DOSBox support.

Right now I'm having a crash on GameHub, but in meantime I would like to clarify some ideas to properly document it:

- There is any reason to GameHub require the config file to run the game on DOSBox? I mean, It could be like the proton, always available to select.

- Considering the config file is needed to enable the DOSBox support for the game, it couldn't be added to "import emulated game"? (showing the dosbox.conf file in the list)

I was testing to change the game executable on game proprieties but it doesn't seems to matter which file is on executable field, since it will execute whatever is on dosbox.conf file instead.

Maybe I can document to put the .config file as executable (or maybe saying to call any file on the game's directory)

**Comment by tkashkin** [comment]
@hlechner 

* There is no particular reason to require config, but it was implemented this way because most DOSBox games (at least on GOG) bundle a few `.conf`s and DOSBox.
`.conf` files seem to be good enough indicator that this game is in fact DOSBox game.
* Game executable is not used because configs already have everything setup.

I'm not sure it's worth to allow to use DOSBox without game-specific configs since most users will probably try to run GOG/Steam games and they usually bundle configs.
If that's not your usecase you can create config manually or run `config -wc game.conf`.

**Comment by neuromancer** [comment]
> I'm not sure it's worth to allow to use DOSBox without game-specific configs since most users will probably try to run GOG/Steam games and they usually bundle configs.
> If that's not your usecase you can create config manually or run config -wc game.conf.

I agree with this. The current approach to detect and use .conf files is as automatic as possible to let users download any GOG game using DOSBox and play it. If the game is not working, you need to tweak the .conf file (or the cue files) to fix the paths.

(I'm the guy that submitted all the [usually annoying] bugs related with DOSBox detection and use in GOG [and sometimes Humble Bundle] games).

**Comment by hlechner** [comment]
My questions were to fully understand how it should work, since I don't have a GOG/Humble game with dosbox on it and my experience were just by running the pure DOS games without dosbox files.

Give me some feedback on following wiki page:
https://github.com/tkashkin/GameHub/wiki/DOSBox-support

**Comment by hlechner** [comment]
the DOSBox GOG and Humble bundle games works the same way?

 I suppose it works this way:
> Click on install, once installed go to properties and enable the DOSBox on compatibility layer.
> 
> If the game do not run on compatibility mode, it will run the DOSBox included on the game itself if do run on compatibility mode it will run on system's DOSBox instead.

Is it right? Maybe I should document it too.

**Comment by tkashkin** [comment]
@hlechner GOG DOSBox games bundle both DOSBox and configs.

If you install Linux version of GOG game, bundled DOSBox will be used by default. You can force compatibility and use DOSBox installed in system.

If you install Windows version of GOG game, compatibility layer will be required. DOSBox should be selected automatically, so usually you just need to run game and it should work.

**Comment by neuromancer** [comment]
Following this discussion, I proposed a new feature (#273) to easily add DOS games that does not have an associated .conf files.
